pkg/controller/client_builder_dynamic.go

```diff
@@ -73,7 +73,7 @@ type DynamicControllerClientBuilder struct {
 	clock clock.Clock
 }
 
-func NewDynamicClientBuilder(clientConfig *restclient.Config, coreClient v1core.CoreV1Interface, ns string) ControllerClientBuilder {
+func NewDynamicClientBuilder(clientConfig *restclient.Config, coreClient v1core.CoreV1Interface, ns string) clientbuilder.ControllerClientBuilder {
```

Comment:
Where is the dynamic client builder being used? Can it also move to `k8s.io/controller-manager`?

Reply:
Ah, I see it's used in kube-controller-manger only. Non-blocking comment in that case but still wondering if it should move there.

Reply:
I am not sure if it is a good move since we also have k8s.io/kube-controller-manager for kube-controller-manager specific staff. Anyway the move might not happen in this PR :)

Reply:
+1. I think we will eventually have to move it over. I believe this is for service account token creation. Eventually a cloud provider is going to want to separate that functionality out from the KCM. Not exactly urgent but I'd be up for seeing it moved preemptively.

Reply:
I could take this up after this PR merged. :)

Reply:
sgtm, thanks :)